### DISCUSSION TITLE
入力モードを「ひらがな」にしても「直接入力」と同じ動作をすることがある。

### DISCUSSION THREAD

**Discussion by hayashida-katsutoshi** [google/mozc#908] [Q&A]
## Description
入力モードを「ひらがな」にしても「直接入力」と同じ動作をすることがあります。

## Steps to reproduce
1. 日本語と英語を、Super+SPACEで切り替えながら入力する。
2. しばらく繰り返していると「mozc」の入力モードがランダムに「直接入力」になる。
3. タスクバーから入力モードを「ひらがな」に切り替える。
4. キー入力する。
5. 日本語(mozc)が選ばれ、入力モードに「ひらがな」が選ばれているのにも関わらず、実際の動作は「直接入力」になる。
再現率は数%くらいです。

## Expected behavior
日本語(mozc)の入力モードを「ひらがな」にしたら、「ひらがな」で入力されて欲しい。

## Actual behavior
「mozc」の入力モードは「ひらがな」が選ばれているのに、実際の動作は「直接入力」になる。

## Screenshots
n/a

## Version or commit-id
2.26.4220.102

## Environment
 - Ubuntu 22.04.4 LTS
 - IBus
 - FeatherPad 1.0.1, Microsoft Edge

## Investigations
 - 英語キーボードを使用。変換、無変換キーはありません。

**Comment by yukawa** [comment]
`2.26.4220.102` is a quite old build, which we published more than 3 years ago. https://github.com/google/mozc/commit/9ba59b64d53365c1fe93c1c245b4ec3e35bdadf0

Until this is confirmed to be an issue that is still reproducible with the latest version of Mozc and is something that can/should be fixed in the Mozc side, let's talk about this in the Discussions pages as we would like to use Issues pages for something actionable.

**Comment by yukawa** [comment]
As far as I know, such an issue has never been reported in other platforms such as macOS and Windows, which makes me suspect the issue you observed was one of the following cases.
* It is an issue in [ibus](https://github.com/ibus/ibus) runtime.
* It is an issue in Gnome (and its extensions)
* It is an issue in `ibus-mozc`

Another thing we should be aware of is that even if the root cause was identified and fixed, you wont' receive the fix automatically as Ubuntu is not a rolling update OS. Here are necessary steps for you to be able to receive the **fix**.

1. Someone decides to take their time to work on this issue.
2. Succeed to reproduce the issue locally, which is most of the cases critical to identify the root cause.
3. Actually identify the root cause.
4. Come up with an idea on how to fix it in which component. The component to be fixed is not necessarily Mozc. It can be any other relevant component such as [ibus](https://github.com/ibus/ibus)
5. Create a proposed fix and get it reviewed by the component owner.
6. Get OK from the component owner to submit the fix.
7. Reach out to the package maintainer in Debian so that a fixed version of component can be imported to Debian unstable repository.
8. Wait until the fixed version of component is actually imported to the Debian unstable repository. This requires an approval from the Debian package maintainer.
9. Wait until the fixed version of component is imported to Ubuntu. If you only use Ubuntu LTS, the earliest version of Ubuntu that could contain the fix is most likely Ubuntu 26.04.
10. Finally you can update your Ubuntu OS to get the issue fixed.

We are currently waiting for the step 1 to happen.
